lib/rubocop/cop/performance/string_replacement.rb

```diff
@@ -0,0 +1,147 @@
+# encoding: utf-8
+
+module RuboCop
+  module Cop
+    module Performance
+      # This cop identifies places where `gsub` can be replaced by
+      # `tr` or `delete`.
+      #
+      # @example
+      #   @bad
+      #   'abc'.gsub('b', 'd')
+      #   'abc'.gsub('a', '')
+      #   'abc'.gsub(/a/, 'd')
+      #   'abc'.gsub!('a', 'd')
+      #
+      #   @good
+      #   'abc'.gsub(/.*/, 'a')
+      #   'abc'.gsub(/a+/, 'd')
+      #   'abc'.tr('b', 'd')
+      #   'a b c'.delete(' ')
+      class StringReplacement < Cop
+        MSG = 'Use `%s` instead of `%s`.'
+        DETERMINISTIC_REGEX = /^[\w\s\-,."']+$/
+        REGEXP_CONSTRUCTOR_METHODS = [:new, :compile]
+
+        def on_send(node)
+          _string, method, first_param, second_param = *node
+          return unless [:gsub, :gsub!].include?(method)
+          return unless second_param && second_param.str_type?
+
+          case first_param.type
```

Comment:
I think the calculation of `first_source` can be extracted into a private method and reused from `autocorrect`.